### ISSUE TITLE
Add support for Spot fleets

### ISSUE THREAD

**Issue by sylvinus** [nchammas/flintrock#138]
See documentation about the feature here:
http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-fleet.html

Adding a `--spot-fleet-request-config` flag that points to the user-supplied `config.json` file would be great!

**Comment by nchammas** [comment]
@sylvinus - Can you summarize the value of this feature to you and contrast it with [auto-scaling groups](http://docs.aws.amazon.com/autoscaling/latest/userguide/WhatIsAutoScaling.html)? I haven't used either feature, but it looks like [you can launch spot instances as part of an auto-scaling group](http://docs.aws.amazon.com/autoscaling/latest/userguide/US-SpotInstances.html), so I'm wondering what the difference is.

**Comment by sylvinus** [comment]
@nchammas I think what Spot fleets add is the notion of `instance weight`, to be able to mix instance types in order to take advantage of lower spot prices and fall back automatically to other launch configurations when prices surge.

I have not used the feature myself yet but it's definitely something I'd test.
